src/lib.rs

```diff
@@ -542,6 +410,43 @@ impl KafkaJsonToDelta {
         Ok(())
     }
 
+    /// Check and seek consumer if rebalance has happened
+    async fn check_rebalance_event(
+        &self,
+        state: &mut ProcessingState,
+    ) -> Result<(), ProcessingError> {
+        let mut partition_assignment = self.partition_assignment.lock().await;
+
+        // TODO if the new assignment is only an addition of partitions we don't need to reset state
+        if partition_assignment.rebalance.is_some() {
+            let partitions = partition_assignment.rebalance.as_ref().unwrap().clone();
```

Comment:
i feel like this unwrap and clone can be avoided if we refactor how we check and reset partition assignments. but since this is not in the hot path, the overhead should be minor, so we can do the clean in future PRs.

Reply:
Okay so this took me a while. Yes I don't need clone, but I couldn't make it compile otherwise. 

The errors I was getting is something like _cannot borrow as the immutable because it's already borrowed a mutable_

Reply:
yeah, to workaround that, we need to combine the rebalance check and `partition_assignment.reset_with` into a single method.